async/logic/Async4BackendRest_DestinationAPI_TopicCheckLogic.java

```diff
@@ -0,0 +1,543 @@
+	    
+	    String destinationUri = new String();
+	    String destinationHttpMethod = new String();
+	    String destinatioRrequestHeader = new String();
+		String destinatioRrequestHeaderKey = new String();
+		String destinatioRrequestHeaderValue = new String();
+	    String destinationRequestBody = new String();
+	    
+        try {
+        	destinationStatus.delete(0, destinationStatus.length());
+        	destinationResponseBody.delete(0, destinationResponseBody.length());
+			executionTimes.delete(0, executionTimes.length());
+			
+        	JSONObject jsonObject = new JSONObject(receivemessage);
+
+			// URLDecodeする
+			destinationUri = URLDecoder.decode(jsonObject.getString("destinationuri"), "utf-8");
+			
+			destinationHttpMethod = jsonObject.getString("destinationhttpmethod");
+			
+			if(receivemessage.contains("destinationrequestheader")) {
+				destinatioRrequestHeader = jsonObject.getString("destinationrequestheader");
+				
+				String[] splitresult = destinatioRrequestHeader.split(":", 0);
+				destinatioRrequestHeaderKey = splitresult[0];
+				
+				destinatioRrequestHeaderValue = splitresult[1];
+				
+			}
+			
+			if(receivemessage.contains("destinationrequestbody")) {
+				// URLDecodeする
+				destinationRequestBody = URLDecoder.decode(jsonObject.getString("destinationrequestbody"), "utf-8");
+				
+			}
+			
+			executionTimes.append(jsonObject.getString("executiontimes"));
+			
+			URL url = new URL(destinationUri);
+			
+			HttpURLConnection urlConnection = null;
+			
+			urlConnection = (HttpURLConnection) url.openConnection();
+			
+			urlConnection.setRequestMethod(destinationHttpMethod);
+			
+			urlConnection.setUseCaches(false);
+			
+			urlConnection.setDoOutput(true);
+			
+			if(receivemessage.contains("destinationrequestheader")) {
+				urlConnection.setRequestProperty(destinatioRrequestHeaderKey, destinatioRrequestHeaderValue);
+			}
+			
+			OutputStreamWriter out = new OutputStreamWriter(
+					new BufferedOutputStream(urlConnection.getOutputStream()));
+			
+			if(receivemessage.contains("destinationrequestbody")) {
+				out.write(destinationRequestBody.toString());
+			}
+			
+			out.close();
+			
+			destinationStatus.append(urlConnection.getResponseCode());
+			
+			BufferedReader br = null;
+			
+			try {
+				// レスポンスを受け取る
+				br = new BufferedReader(new InputStreamReader(urlConnection.getInputStream(), "UTF-8"));
+				
+				String line;
+				
+				while((line = br.readLine()) != null) {
+					destinationResponseBody.append(line);
+				}
+				
+				br.close();
+				
+			// 200番台以外の場合
+			} catch(IOException e) {
+				
+				BufferedReader errbr = null;
+				try {
+					
+					errbr = new BufferedReader(new InputStreamReader(urlConnection.getErrorStream(), "UTF-8"));
+
+					String errline;
+					
+					while((errline = errbr.readLine()) != null) {
+						destinationResponseBody.append(errline);
+					}
+					
+					return GeneralErrorRes(ErrorResponseEnum.E500);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the resource returned by the following method call: *GeneralErrorRes*. The resource is returned at line 332. There are other execution paths that don&#39;t close the resource or return it, for example, when BufferedReader.readLine throws an exception. Ensure the object returned by *GeneralErrorRes()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).